javascript/ql/src/semmle/javascript/Regexp.qll

```diff
@@ -404,3 +407,87 @@ module RegExpPatterns {
     result = "(?:com|org|edu|gov|uk|net|io)(?![a-z0-9])"
   }
 }
+
+/**
+ * Gets a node whose value may flow (inter-procedurally) to a position where it is interpreted
+ * as a part of a regular expression.
+ */
+private DataFlow::Node regExpSource(DataFlow::Node re, DataFlow::TypeBackTracker t) {
+  t.start() and
+  re = result and
+  isInterpretedAsRegExp(result)
+  or
+  exists(DataFlow::TypeBackTracker t2, DataFlow::Node succ | succ = regExpSource(re, t2) |
+    t2 = t.smallstep(result, succ)
+    or
+    any(TaintTracking::AdditionalTaintStep dts).step(result, succ) and
+    t = t2
```

Comment:
`t = t2.continue()` would be slightly more correct, I believe (though I doubt there is any practical difference).

Reply:
This is an unchanged move from #1211.

The tests in that PR shows a semantic difference.
The following `NOT OK` line is *not* flagged if `t = t2.continue()` is used.

```
	let domains = [ { hostname: 'test.example.com$' } ];  // NOT OK
	function convert2(domain) {
		return new RegExp(domain.hostname);
	}
	domains.map(d => convert2(d));
```

Reply:
Oh, right, forgot about this one.